Feat/back/#12/create an experience event

arashmoradi81 opened the pull request.

arashmoradi81 committed:
Merge branch 'feat/back/add_UserRegisterService_and_BaseRequest' of https://github.com/zeinabmontazeri/Options

aghaSeyed committed:
CRUD on Admin\CategoryController

arashmoradi81 committed:
Merge branch 'main' of https://github.com/zeinabmontazeri/Options

ElmiraBehnava committed:
feat(back): #6 get experienc events

ElmiraBehnava committed:
fix(back): #6 remove comments

ElmiraBehnava committed:
fix(back): formatter on framework.yml file

aghaSeyed committed:
provide json response according to jsend response standard

ElmiraBehnava committed:
fix(back): improve structre

mehdad committed:
Fix user entitiy issues

mehdad committed:
Implement required validation and add new custom exception for validation

mehdad committed:
Add new controller and service  for register user

mehdad committed:
Make Role not nullable

mehdad committed:
Change Role Enums
Add logic for host registration

mehdad committed:
Add base route to auth address

aghaSeyed committed:
add features mentioned in ticket #11 ,some exceptions are handled manually.

arashmoradi81 committed:
rebase #12 with dev

arashmoradi81 committed:
add feature create an experience event

aghaSeyed committed:
add id of created experience to data's response

arashmoradi81 committed:
Update EventService.php

arashmoradi81 committed:
Update EventService.php

mehdad committed:
Add logout mechanism to endpoint

s-marashi committed:
feat: #3-Login

ElmiraBehnava committed:
feat(back): # remove an order

ElmiraBehnava committed:
fix(back): #5 fix some issues

mehdad committed:
Revert "Add logout mechanism to endpoint"

This reverts commit 321df2caa640e42297b5fab0e8f730495a9ee94a.

s-marashi committed:
fix: #3 - Login
- Set content of role in token
- Handle unprovided phoneNumber and password
- Reformat responses to follow convention
- Reformat listeners to be more readable and uniform
- Refactor loginChecker to extract exception throwing from service
- Handle response for failed authentication
- Handle response for successful authentication

ElmiraBehnava committed:
fix(back): remove an order and just copied right version of entities

ElmiraBehnava committed:
fix(back): fix confilicts

s-marashi commented on src/Request/EventRequest.php:
```diff
@@ -0,0 +1,42 @@
+
+namespace App\Request;
+
+use App\Entity\Experience;
```
Please consider removing unnecessary use statements.

ElmiraBehnava committed:
fix(back): add use

s-marashi commented on src/Service/EventService.php:
```diff
@@ -0,0 +1,34 @@
+    public function create(EventRequest $request, EventRepository $repository, Experience $experience): string
+    {
+        if ($request->errors){
+            throw new \Exception($request->errors);
```
It would be much better if we throw a specific exception instead of a general one.

s-marashi commented on src/Request/ValidateRequestTrait.php:
```diff
@@ -7,7 +7,10 @@ public function populate(array $fields): void
         foreach ($fields as $field => $value) {
             if (property_exists($this, $field)) {
-                $this->{$field} = $value;
+                if (preg_match('/^\d{4}-\d\d-\d\d( \d\d:\d\d:\d\d)?$/', $value))
```
This type conversion is so fragile please consider a transformation step before validation

arashmoradi81 replied:
این رو من خیلی تست کردم. برای validation startAt که بیشتر از زمان حال باشه جز اینجا که به تاریخ تبدیل میکنیم جای دیگه نمیشه این کار را انجام داد. پیشنهادی بود حتما بگو

s-marashi commented on src/Controller/Host/CreateEventController.php:
```diff
@@ -0,0 +1,36 @@
+use App\Service\EventService;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
```
Please consider removing unnecessary use statements.

s-marashi commented on src/Controller/Host/CreateEventController.php:
```diff
@@ -0,0 +1,36 @@
+
+class CreateEventController extends AbstractController
+{
+    #[Route('/host/{id}/create/event', name: 'app_host_create_event', methods: ['POST'])]
```
The semantic of URI seems to be wrong. I think it means that we are trying to load a host with id given in the URI but the logic is searching for a experience with that id please change route.

s-marashi commented on src/Controller/Host/CreateEventController.php:
```diff
@@ -0,0 +1,36 @@
+class CreateEventController extends AbstractController
+{
+    #[Route('/host/{id}/create/event', name: 'app_host_create_event', methods: ['POST'])]
+    public function createEvent(Experience $experience, EventRepository $repository, EventRequest $request, EventService $eventService)
```
Please consider injecting event's serivce (i.e. EventRepository and maybe Experience) dependencies into its constructure.

s-marashi commented on src/Controller/Host/CreateEventController.php:
```diff
@@ -0,0 +1,36 @@
+        try {
+            $msg = $eventService->create($request, $repository, $experience);
+            return new JsonResponse([
+                'data' => [],
```
Please consider removing unnecessary use statements.

s-marashi commented on src/Request/EventRequest.php:
```diff
@@ -0,0 +1,42 @@
+    #[Assert\NotNull]
+    public ?int $capacity = 0;
+
+    #[Assert\NotNull]
```
Please consider applying stronger validation on all fields

aghaSeyed committed:
feat(back) : add exceptionListener

s-marashi commented on src/Request/BaseRequest.php:
```diff
@@ -28,8 +28,17 @@ public function validate(): void
         if (count($errors) > 0) {
-            $errorsString = (string)$errors;
-            throw new Exception($errorsString);
+            $this->makeError((string)$errors);
```
Please consider passing errors as array to prevent following explode.

s-marashi commented on src/Service/EventService.php:
```diff
@@ -0,0 +1,34 @@
+use App\Entity\Event;
+use App\Entity\Experience;
+use App\Repository\EventRepository;
+use App\Repository\ExperienceRepository;
```
Please consider removing unnecessary use statments.

s-marashi committed:
fix: #3-Login
- A better error handling strategy for missing phoneNumber and password

mehdad committed:
Add lexik and change security to support login and register

s-marashi committed:
feat: #7:Access Control
- It defines AcceptableRoles
- It checks if login user is present in acceptable roles
- It Handle invalid provided roles
- It handle unauthorized access

s-marashi committed:
fix: #7 - Access control
- Make acceptable roles a unique array.

ElmiraBehnava committed:
fix(back): fix gitignore

ElmiraBehnava committed:
imp(back): add type into query

ElmiraBehnava committed:
fix(back): .env fixed

aghaSeyed committed:
fix(back) : ExperienceCollection changed

ElmiraBehnava committed:
fix(back): refactor name

ElmiraBehnava committed:
fix(back): add use for user

ElmiraBehnava committed:
fix(back): user entity merged to main'

ElmiraBehnava committed:
fix(back): remane controller

mehdad committed:
Check birthday on registration

ElmiraBehnava committed:
fix(back): conflicts

mehdad committed:
Add host if user registers as Host
Change Gender type to string

mehdad committed:
Add Host if user registers as host
Change Geneder Type

mehdad committed:
Merge branch 'feat-#2-Register' of https://github.com/zeinabmontazeri/Options into feat-#2-Register

mehdad committed:
Add new method to user repository to check user existance

ElmiraBehnava committed:
fix(back): fix user entity and add checking status is dreaf

ElmiraBehnava committed:
fix(back): fix some issues

aghaSeyed committed:
feat(back) : Event has rigisteredUsers , User has enum gender , Order has enum status changed.

AppFixtures create fake data for testing postman query

aghaSeyed committed:
feat(back) : Event has rigisteredUsers , User has enum gender , Order has enum status changed.

AppFixtures create fake data for testing postman query

aghaSeyed committed:
fix(back) : merged with main

aghaSeyed committed:
Merge branch 'dev' into feat/back/#14/manage_category

aghaSeyed committed:
Merge pull request #20 from zeinabmontazeri/feat/back/#14/manage_category

Feat/back/#14/manage category

mehdad committed:
Change registration service to have injection in write times

s-marashi committed:
fix: #3-Login
- Return user id in response of login

mehdad committed:
Save Role as String

aghaSeyed committed:
fix(back) : request change has done!

aghaSeyed committed:
Merge branch 'dev' into feat/back/#11/create_an_experiment

aghaSeyed committed:
Merge pull request #21 from zeinabmontazeri/feat/back/#11/create_an_experiment

Feat/back/#11/create an experiment

arashmoradi81 committed:
delete email from user entity

arashmoradi81 committed:
delete set email in users

arashmoradi81 committed:
add getCapacity function

aghaSeyed committed:
Merge branch 'dev' into feat#3/login

aghaSeyed committed:
Merge pull request #22 from zeinabmontazeri/feat#3/login

Feat#3/login

mehdad committed:
Change roles back to array in user registration

Commit:
Merge branch 'main' into feat-#2-Register

mahnazzare committed:
fix(back):get ategory with findall

ElmiraBehnava committed:
fix(back): merge logic of branch #6 and #5

ElmiraBehnava committed:
fix(back): just copy user entity from main

ElmiraBehnava committed:
fix(back): add experienc event service

ElmiraBehnava committed:
fix(back): conflicts

ElmiraBehnava committed:
fix(back): fix conflicts

mahnazzare committed:
fix(back):add method get to function

aghaSeyed committed:
feat(back) : #17 add host report.

aghaSeyed committed:
Revert "feat(back) : #17 add host report."

This reverts commit 570fb30b8e82487028158af430c119d66844065a.

arashmoradi81 committed:
Merge branch 'main' into feat/back/#12/create_an_experience_event

s-marashi commented:
This is going to be a review

arashmoradi81 committed:
rebase #12 with dev

mehdad committed:
Merge Dev With Register Branch

mehdad committed:
Merge Dev Into Logout Branch

mehdad committed:
Make Register controller to use global exception handler
Change global exception handler to support validation errors

mehdad committed:
Merge branch 'dev' into feat-#4-Logout

# Conflicts:
#	src/EventListener/ExceptionSubscriber.php

mehdad committed:
Change login success priority to have refresh token in event body
Add refresh token to data parameter of successful login object

arashmoradi81 committed:
Merge branch 'dev' into feat/back/#12/create_an_experience_event

aghaSeyed committed:
feat(back) : #17 add host report.

ElmiraBehnava committed:
fix(back): fix confilicts

aghaSeyed committed:
fix(back) : #17 delete pending status of order

aghaSeyed committed:
Merge branch 'dev' of https://github.com/zeinabmontazeri/Options into dev

mehdad committed:
Add refresh token entity

mehdad committed:
Merge remote-tracking branch 'origin/dev' into dev

ElmiraBehnava committed:
feat(back): add payment entity and repository

s-marashi committed:
feat(back): #10 Checkout: Defined Task description for payments.

s-marashi committed:
feat(back): #10 Checkout: make feilds readonly

s-marashi committed:
feat(back): #3: Login fix dependency on request body

mehdad committed:
Fix branch issue with requests

s-marashi committed:
fix: #3 - Early rejection in _LoginChecker_
- As toArray method force request to have a body by early rejecting
requests from routes other than auth.login prevents this inforcement.

ElmiraBehnava committed:
fix(back): fix base request

ElmiraBehnava committed:
fix(back): namesapce

ElmiraBehnava committed:
fix(back): add role above controller

ElmiraBehnava committed:
fix(back): fix confilicts

s-marashi committed:
Merge branch 'feat/back/#10/checkout' of https://github.com/zeinabmontazeri/Options into feat/back/#10/checkout

ElmiraBehnava committed:
fix(back): fix namesapace

s-marashi committed:
feat: #38 - Project setup makefile

s-marashi committed:
Merge branch 'dev' into feat#38/Makefile

ElmiraBehnava committed:
Merge branch 'dev' into feat/back/#10/checkout

ElmiraBehnava committed:
fix(back): add accaptable roles and fix urls and some indents"

ElmiraBehnava committed:
Merge branch 'dev' into feat/back/#9/remove_an_order

ElmiraBehnava committed:
Merge pull request #30 from zeinabmontazeri/feat/back/#9/remove_an_order

Feat/back/#9/remove an order

ElmiraBehnava committed:
Merge branch 'dev' into feat/back/#9/remove_an_order

ElmiraBehnava committed:
Merge pull request #31 from zeinabmontazeri/feat/back/#9/remove_an_order

Feat/back/#9/remove an order

ElmiraBehnava committed:
fix(back): improvement

ElmiraBehnava committed:
fix(back): remove unsed import

ElmiraBehnava committed:
Merge pull request #32 from zeinabmontazeri/feat/back/#6/get_experience_events

Feat/back/#6/get experience events

ElmiraBehnava committed:
fix(back): add api/v1/ prefix

ElmiraBehnava committed:
fix(back): fix roles

aghaSeyed committed:
fix(back) : adding factory design pattern.

refactor all branches 11 14 17

aghaSeyed committed:
Merge branch 'dev' into feat/back/#43/refactor_11_14_17

ElmiraBehnava committed:
fix(back): apply dto factory to branch #6 and #5

ElmiraBehnava committed:
fix(back): resolve confilicts

ElmiraBehnava committed:
fix(back): apply dto factory to branch #6 and #5

ElmiraBehnava committed:
Merge pull request #33 from zeinabmontazeri/feat/back/#9/remove_an_order

fix(back): fix roles

arashmoradi81 committed:
add transformer in validation with factory design pattern

ElmiraBehnava committed:
fix(back): resolve conflicts

ElmiraBehnava committed:
Merge pull request #34 from zeinabmontazeri/feat/back/add_dto_factory

Feat/back/add dto factory

arashmoradi81 committed:
Merge branch 'dev' of https://github.com/zeinabmontazeri/Options into feat/back/#12/create_an_experience_event

 Conflicts:
	composer.lock
	src/EventListener/ExceptionSubscriber.php
	src/Request/EventRequest.php
	src/Request/UserRegisterRequest.php
	src/Request/ValidateRequestTrait.php
	src/Service/EventService.php
	src/Service/UserRegisterService.php

arashmoradi81 committed:
add access for create event
